packages/bento-design-system/src/TextField/createTextField.tsx

```diff
@@ -31,34 +38,62 @@ export function createTextField(config: InputConfig, { Field }: { Field: FieldTy
       inputRef
     );
 
+    const [showPassword, setShowPassword] = useState(false);
+    const passwordIcon = showPassword
+      ? config.passwordHideIcon ?? IconEye
+      : config.passwordShowIcon ?? IconEyeClosed;
+
+    const iconProps: IconProps = { size: config.passwordIconSize ?? 24 };
+
+    const type = props.type === "password" && !showPassword ? "password" : "text";
+
     return (
       <Field
         {...props}
         labelProps={labelProps}
         assistiveTextProps={descriptionProps}
         errorMessageProps={errorMessageProps}
       >
-        <Box
-          as="input"
-          type={props.type || "text"}
-          ref={inputRef}
-          {...inputProps}
-          // NOTE(gabro): this is to please TS, since the inputProps type is very broad
-          color={undefined}
-          width={undefined}
-          height={undefined}
-          borderRadius={config.radius}
-          paddingX={config.paddingX}
-          paddingY={config.paddingY}
-          className={[
-            inputRecipe({ validation: validationState || "notSet" }),
-            bodyRecipe({
-              color: props.disabled ? "disabled" : "default",
-              weight: "default",
-              size: config.fontSize,
-            }),
-          ]}
-        />
+        <Box position="relative" display="flex">
+          <Box
+            as="input"
+            ref={inputRef}
+            {...inputProps}
+            type={type}
+            // NOTE(gabro): this is to please TS, since the inputProps type is very broad
+            color={undefined}
+            width={undefined}
+            height={undefined}
+            borderRadius={config.radius}
+            paddingX={config.paddingX}
+            paddingY={config.paddingY}
+            className={[
+              inputRecipe({ validation: validationState || "notSet" }),
+              bodyRecipe({
+                color: props.disabled ? "disabled" : "default",
+                weight: "default",
+                size: config.fontSize,
+              }),
+            ]}
+            style={{ paddingRight: rightAccessoryWidth, flexGrow: 1 }}
+          />
+          {props.type === "password" && (
+            <Box
+              ref={rightAccessoryRef}
+              position="absolute"
+              display="flex"
+              justifyContent="center"
+              alignItems="center"
+              paddingX={16}
+              top={0}
+              bottom={0}
+              right={0}
+              onClick={() => setShowPassword((prevValue) => !prevValue)}
+            >
+              {passwordIcon(iconProps)}
+            </Box>
```

Comment:
Can you use `IconButton` here? `IconButton` gives you a few features such as bigger target area, proper hover, keyboard management and accessibility via an aria-label.

You probably still need the `Box` for positioning the element, but you can remove the `onClick` from it

Reply:
A label is required for IconButton. What should be set?

Reply:
right, those should come from default messages and be customizable. You can observe the same pattern in `Chip`, see how the `dismissButtonLabel` is handled (optional prop on the component, mandatory in `DefaultMessages`).

This will also require to update the `defaultMessages` in a few spots (you can just use `"Hide password"` / `"Show password"`)